### PR TITLE
Emag functionality on firelocks

### PR DESCRIPTION
**Pull request by eplgr**
https://puu.sh/tnVuK/9f34dcb86d.webm

Currently, emags can't open firelocks: you get the message "access denied", which doesn't make sense. 

Also, remember that time when you were a nuke op with a bag full of weapons and had no space for tools? Because no one carries a crowbar as a nuke op. Then the rest of your team bombed the whole station, firelocks were on high alert, and you found yourself stuck there forever with no way out. You kept trying to use your emag and thinking to yourself "WHY!".

Also, this is my first PR. 

:cl: 
 * rscadd: Emags can now hack open firelocks
 * rscadd: Added sprites for firelocks with sparks

### CODE REVIEW

**Comment by Kurfursten** [comment]
This is functional but probably not what people are expecting. Most people would probably assume this would force it to stay in an open state, like what it does to a door. This will just open it temporarily, as if you crowbared it.

**Comment by aceedex** [comment]
some things out to remain a hinderance to the all mighty emag

**Comment by eplgr** [comment]
@Kurfursten 
I understand that and also spent some time thinking about that issue. The real problem there is that there is no way to actually build and/or fix firelocks. You can only weld and unweld them.
And implementying a code for firelocks to be broken/fixable and probably buildable is currently too much for me.

Also just plain getting an "Access Denied" message is just...wrong. Very wrong. That's actually unexpected and "harmful" to the gameplay.

**Comment by Probe1** [comment]
We can't repair firelocks can we?

We can't even build them.

So having it permanently in a bad state is gonna need way more code to repair them.  This will do.

**Comment by eplgr** [comment]
After some discussion, I decided I'll be doing some changes so it stays open and "breaks". And sparks.

**Comment by SonixApache** [comment]
It makes sense for it to just open temporarily when emagged. Using a normal ID on a firelock does the same thing.

IMO the only thing you should try to do is make sure it doesn't list the emagger in the examination list that shows who've opened the firelock during an emergency.

File: code/game/machinery/doors/firedoor.dm [hunk 1]
```diff
@@ -233,7 +233,7 @@ var/global/list/alert_overlays_global = list()
 			update_icon()
 			return
 
-	if( iscrowbar(C) || ( istype(C,/obj/item/weapon/fireaxe) && C.wielded ) )
+	if( iscrowbar(C) || isEmag(C) || ( istype(C,/obj/item/weapon/fireaxe) && C.wielded ) )
```
Comment (ComicIronic) [inline code/game/machinery/doors/firedoor.dm]: Why is this before the welded check? An emag can't do anything to a welded door.

**Comment by eplgr** [comment]
Alright. I finished it. 
I think.

**Comment by gbasood** [comment]
So they'll be stuck open permanently after those last few commits, then?
Is there any way to repair them?

**Comment by eplgr** [comment]
I wasn't sure whether I should explain or not how to "fix" them in the changelog. The idea is that after the sparks, the firelocks are "welded". You can just use a welder to make them functional again.

**Comment by PJB3005** [comment]
Honestly just sparking and opening without leaving a record sounds fine.

The welder thing I'm not a fan of.

**Comment by eplgr** [comment]
But if you force_open it, it'd just close shortly after. It doesn't feel right. 
I'll see what I can do about the record thing.

**Comment by PJB3005** [comment]
Yeah I don't think it feels right either but causing it to be welded shut is even weirder.

**Comment by Shadowmech88** [comment]
Is there a way to have it stay open until someone touches it with their hand or an ID or a crowbar or something? So that it's not *welded* open, but it won't drop again without player intervention?

**Comment by PJB3005** [comment]
Without extra code probably not.

**Comment by Shadowmech88** [comment]
I guess having them simply open is fine for now, if he doesn't feel up to writing more code into firelocks yet.

**Comment by eplgr** [comment]
A colaborator, I think, suggested it doesn't really need to follow logic. And also that how can you know the sparks are not hot enough to cause the firelock to melt a little, welding itself? Etcetera. I bought the story. I don't think it's that bad.

Anyway, I will keep researching how to improve the code.

**Comment by eplgr** [comment]
I reviewed and tested a few things with the records. If I understood correctly, if you examine a firelock that was emagged, it won't show who opened it. The whole thing is probably bugged, though. It doesn't show either if it was opened with a crowbar. The names ONLY appear if you touch it directly with your hands. To be clear: that's with the bleeding edge code.

Being said that, I think this is as far as I go with this PR. I like its current state.

**Pull request merged**